data/blog/2025/2025-02-13-mezzio101-creating-pages-and-modules.md

```diff
@@ -0,0 +1,198 @@
+
+The new handler requires a Factory.
+Let's create the `src/App/src/Handler/AboutPageHandlerFactory.php` file.
+Its content is very similar to the `HomePageHandlerFactory.php` file, but returns the `AboutPageHandler` class instead.
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Handler;
+
+use Mezzio\Router\RouterInterface;
+use Mezzio\Template\TemplateRendererInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+use function assert;
+
+class AboutPageHandlerFactory
+{
+    public function __invoke(ContainerInterface $container): RequestHandlerInterface
+    {
+        $router = $container->get(RouterInterface::class);
+        assert($router instanceof RouterInterface);
+
+        $template = $container->has(TemplateRendererInterface::class)
+            ? $container->get(TemplateRendererInterface::class)
+            : null;
+        assert($template instanceof TemplateRendererInterface || null === $template);
+
+        return new AboutPageHandler($container::class, $router, $template);
+    }
+}
+```
+
+#### The ConfigProvider file
+
+The factory needs to be added in the `src/App/src/ConfigProvider.php` file.
+The location to add the new entry is in the `getDependencies` method, under the `factories` key, like we did below.
+You can replace the current `getDependencies` method with this one:
+
+```php
+    public function getDependencies(): array
+    {
+        return [
+            'invokables' => [
+                Handler\PingHandler::class => Handler\PingHandler::class,
+            ],
+            'factories'  => [
+                Handler\HomePageHandler::class  => Handler\HomePageHandlerFactory::class,
+                Handler\AboutPageHandler::class => Handler\AboutPageHandlerFactory::class,
+            ],
+        ];
+    }
+```
+
+#### The routing
+
+Finally, add the new handler to the `config/routes.php` file.
+On the bottom of the file we already have 2 entries created by the Mezzio Skeleton Installer.
+You can check the Handler names to tell what each of them is for.
+
+For the new page, we need to append the line for the `AboutPageHandler` to that list, like we have done below.
+Feel free to replace the return statement with this one:
+
+```php
+return static function (Application $app, MiddlewareFactory $factory, ContainerInterface $container): void {
+    $app->get('/', App\Handler\HomePageHandler::class, 'home');
+    $app->get('/api/ping', App\Handler\PingHandler::class, 'api.ping');
+    $app->get('/about', App\Handler\AboutPageHandler::class, 'about');
+};
+```
+
+#### The result
+
+You are done!
+Next make sure to have your web server started by running this command in your project folder `php -S 0.0.0.0:8080 -t public`.
+To visit your new page, type this url in the browser `http://localhost:8080/about`.
+
+### Creating a new module
+
+If you want to organize your pages in several modules, Mezzio provides an easy method to create the folder structure.
```

Comment:
I do not understand this part? If it is a complete tutorial that the reader is supposed to follow and practise, then this part is somehow superfluous. Will modules be used in this tutorial? If not, then this part should be removed or should be placed at the end as a possible alternative implementation.

Reply:
a new module isn't really needed for a new handler used to list a static page, but i mentioned modules to highlight what they're good for in more complex applications

Reply:
> …but i mentioned modules to highlight what they're good for in more complex applications

Put it to the end like performance optimisation, alternatives or tips on additional helpful tools. Or create separate section as a "box for a tip" and add content that matches the current article.

Reply:
I'm now considering your first comment on the modules: to remove this section altogether. Would it be ok to create another article that focuses only on modules? I only worry that it's going to be too short.